.github/workflows/release.yml

```diff
@@ -6,19 +6,51 @@ on:
       - 'v*'
 
 jobs:
-  build:
+  ci:
+    name: CI
+    uses: ./.github/workflows/ci.yml
+
+  docs:
+    name: Verify Docs Build
+    uses: beeware/.github/.github/workflows/docs-build-verify.yml@main
+    secrets: inherit
+    with:
+      project-name: "beeware"
+      project-version: ${{ github.ref_name }}
+
+  release:
     name: Create Release
+    needs: [ ci, docs ]
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      # This permission is required for trusted publishing.
+      id-token: write
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-      - name: Create Release
-        id: create_release
-        uses: actions/create-release@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set build variables
+        run: |
+          echo "VERSION=${GITHUB_REF_NAME#v}" >> $GITHUB_ENV
+
+      - name: Set up Python
+        uses: actions/setup-python@v5.0.0
+        with:
+          python-version: "3.x"
+
+      - name: Get packages
+        uses: actions/download-artifact@v4.1.0
         with:
-          tag_name: ${{ github.ref }}
-          release_name: Release ${{ github.ref }}
+          name: ${{ needs.ci.outputs.artifact-name }}
```

Comment:
The `artifact-name` [output](https://github.com/beeware/briefcase/blob/ee60e2f807c8a4461e772f7697247632d146259c/.github/workflows/ci.yml#L9) will need to be added to the `ci` workflow.